File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;


import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务层接口
 */
public interface BrandService {

    /**
     * 返回全部列表
     */

    public List<TbBrand> findAll();

    /**
     * 品牌分页
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    /**
     * 保存新建品牌
     * @param tbBrand
     */
    public void add (TbBrand tbBrand);

    /**
     * 修改品牌
     * @param tbBrand
     */
    public void update(TbBrand tbBrand);

    /**
     *根据id获取实体
     * @param id
     * @return
     */
    public TbBrand findOne(long id);

    /**
     * 批量删除
     * @param ids
     */
    public void delete(long[] ids);

    /**
     * 条件查询
     * @param num
     * @param size
     * @return
     */
    PageResult search(int num, int size,TbBrand tbBrand);

    List<Map>findOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    //读取列表数据绑定到表单中
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    };

    //分页查询
    this.findPage=function (num,size) {
        return  $http.get('../brand/findPage.do?num='+num+'&size='+size);
    };

    //保存方法
    this.add=function (entity) {
        return  $http.post('../brand/add.do',entity);
    };

    //更新方法
    this.update=function (entity) {
        return  $http.post('../brand/update.do',entity);
    };

    //获取实体
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    };

    //批量删除
    this.delete=function (selectIds) {
        return  $http.get('../brand/delete.do?ids='+selectIds);
    };
    //多条件查询
    this.search=function (num,size,searchentity) {
        return $http.post('../brand/search.do?num='+num+'&size='+size,searchentity);
    };
    //品牌下拉列表
    this.findOptionList=function () {
        return  $http.get('../brand/findOptionList.do');

    }
});
